File: app-aula/lib/models/tarea.dart
class Tarea {
  final int id;
  final String titulo;
  final String? contenido;
  final String? fechaEntrega;
  final int? puntos;
  final int cursoId;
  final int? seccionId;

  Tarea({
    required this.id,
    required this.titulo,
    this.contenido,
    this.fechaEntrega,
    this.puntos,
    required this.cursoId,
    this.seccionId,
  });

  factory Tarea.fromJson(Map<String, dynamic> json) {
    return Tarea(
      id: json['id'],
      titulo: json['titulo'],
      contenido: json['contenido'],
      fechaEntrega: json['fecha_entrega'],
      puntos: json['puntos'],
      cursoId: json['curso_id'],
      seccionId: json['seccion_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'titulo': titulo,
      'contenido': contenido,
      'fecha_entrega': fechaEntrega,
      'puntos': puntos,
      'curso_id': cursoId,
      'seccion_id': seccionId,
    };
  }
}

File: backend-laravel/app/Models/Tarea.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Tarea extends Model
{
    protected $fillable = [
        'titulo',
        'contenido',
        'fecha_entrega',
        'puntos',
        'curso_id',
        'seccion_id',
        'modulo_master_id',
    ];

    public function curso(): BelongsTo
    {
        return $this->belongsTo(Curso::class);
    }

    public function moduloMaster(): BelongsTo
    {
        return $this->belongsTo(ModuloMaster::class);
    }

    public function seccion(): BelongsTo
    {
        return $this->belongsTo(Seccion::class);
    }

    public function entregas(): HasMany
    {
        return $this->hasMany(Entrega::class);
    }
}
